src/AzureDevOpsPolicyConfigurator/Logic/PolicyExecuterBase.cs

```diff
@@ -0,0 +1,282 @@
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Creates a policy in TFS.
+        /// </summary>
+        /// <param name="policyClient">Policy client</param>
+        /// <param name="types">Types</param>
+        /// <param name="projectId">Team project id</param>
+        /// <param name="repository">Git repository</param>
+        /// <param name="currentPolicy">Branch policy</param>
+        /// <param name="policy">Policy</param>
+        protected abstract void CreatePolicy(
+            PolicyHttpClient policyClient,
+            IEnumerable<PolicyType> types,
+            Guid projectId,
+            GitRepository repository,
+            BranchPolicies currentPolicy,
+            Policy policy);
+
+        /// <summary>
+        /// Updates a policy in TFS.
+        /// </summary>
+        /// <param name="policyClient">Policy client</param>
+        /// <param name="types">Types</param>
+        /// <param name="projectId">Team project id</param>
+        /// <param name="repository">Git repository</param>
+        /// <param name="currentPolicy">Branch policy</param>
+        /// <param name="policy">Policy</param>
+        /// <param name="tfsPolicy">Tfs policy</param>
+        protected abstract void UpdatePolicy(
+            PolicyHttpClient policyClient,
+            IEnumerable<PolicyType> types,
+            Guid projectId,
+            GitRepository repository,
+            BranchPolicies currentPolicy,
+            Policy policy,
+            PolicyConfiguration tfsPolicy);
+
+        /// <summary>
+        /// Deletes a policy in TFS.
+        /// </summary>
+        /// <param name="policyClient">Policy client</param>
+        /// <param name="projectId">Team project id</param>
+        /// <param name="policy">Policy</param>
+        protected abstract void DeletePolicy(PolicyHttpClient policyClient, Guid projectId, PolicyConfiguration policy);
+
+        private void HandleChanges(
+            bool allowDeletion,
+            PolicyHttpClient policyClient,
+            Guid projectId,
+            GitRepository repository,
+            IEnumerable<BranchPolicies> relevantPolicies,
+            IEnumerable<PolicyConfiguration> tfsPolicies,
+            IEnumerable<PolicyType> types)
+        {
+            List<int> handledTfsPolicies = new List<int>();
+
+            foreach (var currentPolicy in relevantPolicies)
+            {
+                this.HandleBranchChanges(policyClient, projectId, repository, tfsPolicies, types, handledTfsPolicies, currentPolicy);
+            }
+
+            this.RemoveNoMatchTfsPolicies(allowDeletion, policyClient, projectId, tfsPolicies, handledTfsPolicies);
+        }
+
+        private void HandleBranchChanges(
+            PolicyHttpClient policyClient,
+            Guid projectId,
+            GitRepository repository,
+            IEnumerable<PolicyConfiguration> tfsPolicies,
```

Comment:
Use AzureDevOps instead of TFS, as on-prem product is now also called Azure DevOps Server